changelog/next/bug-fixes/4527--precise-parsers.md

```diff
@@ -0,0 +1,2 @@
+We fixed various cases where values would not be properly parsed as
+typed data and remained and remained strings instead.
```

Comment:
```suggestion
We fixed various cases where values would not be properly parsed as
typed data and remained strings instead.
```

Are you talking about the duration-IP parser ordering here, or what is this?

Reply:
I think you resolved this on accident. The changelog still looks the same and my question unanswered.

Reply:
Yup, that was an accident. 

One specific case this applies to was the value `2^63` in the `parse leef over syslog`, which wasnt properly parsed and remained a string. (And thus obscured a bug in the integral parsers...)

Reply:
Unresolving this once more.

Besides the `2^63` thing, what does this refer to? You added "edge [cases]" and that "no input data was lost, but no valuable type information was gained either" afterwards. I appreciate the attempt to qualify the impact of the bug. However, the changelog still reads as if there were several instances of this misparsing, and while the second sentence is funny to read, it does not give any new information. Hence, it'd be great if you could help me understand what this refers to (perhaps we can improve the changelog, but it's also helpful for the review).

Reply:
Essentially all parsers I touched will now attempt to parse all fields as typed data, where the previously may not have  (see above).

I dont have any other concrete examples at hand, but at least the `parse leef`  just retained fields as strings whenever they "failed" to parse. 

This would also be triggered by the parser order issues we discovered recently. Those would fail to parse, because they wouldnt consume the entire string and thus the result would remain a string.

I am fairly certain we also discovered this happening for some timestamps/IPs.